service/scaladsl/api/src/main/scala/com/lightbend/lagom/scaladsl/api/AdditionalConfiguration.scala

```diff
@@ -15,6 +15,9 @@ import play.api.Configuration
  * can control which order this configuration gets applied by changing the order in which traits are mixed together.
  */
 trait ProvidesAdditionalConfiguration {
+  @deprecated(message = "prefer `additionalConfiguration(initialConfiguration: Configuration)`", since = "1.6.0")
```

Comment:
I don't think we need to deprecate this method. We can _just_ add `def additionalConfiguration(initialConfiguration: Configuration)`.

We must carefully document and test the precedence between the two or even consider suggesting users to not mix both overrides (IMHO users should be instructed to prefer using `def additionalConfiguration`).

Reply:
that makes sense. i don't see a reason you'd need to use both signatures in one integration, but so long as the precedence is documented it would seem safe to do it. in my mind it makes most sense to have the version which receives initial configuration get evaluated first, but open to thoughts.

Reply:
> in my mind it makes most sense to have the version which receives initial configuration get evaluated first

Hmmm, interesting. I would have thought the opposite was more convenient. Let me share my reasoning:

1. load `initialConfig`
2. load the static `additionalConfig`
3. take the merged `initialConfig` with `additionalConfig` as argument to the new `additionalConfig(...)` method that can then do a final adjustment (override, trim, etc...) to the `Config` the application will use.

Does that make sense?

Reply:
ah, i did not think to pass the initial `additionalConfig` into the new method. that seems like it could be useful.